doc/proposals/scorecard-custom-tests.md

```diff
@@ -0,0 +1,403 @@
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  labels:
+    vendor: redhat
+    test: mycustomtest
+    testsuite: custom
+spec:
+  replicas: 4
+```
+
+The end user could run that test with this command:
+```
+operator-sdk scorecard -o text --selector=test=mycustomtest
+```
+The scorecard would look at on-disk custom tests as well as tests referenced by URL and perform label selection/matching to determine which tests would be executed.
+
+#### Sample Scorecard Usage
+
+Here is how an SDK user would work with scorecard after this enhancement is implemented:
+
+ * Develop Custom Test Steps
+ * Develop Custom Test Assertions
+ * Configure Scorecard to Locate Custom Tests
+ * Execute Scorecard Specifying Custom Test Label(s)
+ * Scorecard Aggregates Custom Test Output/Results
+ * Scorecard Displays Test Results
+
+#### Phase 2
+
+For consideration, with the above custom test capability, there is benefit to Red Hat in migrating the existing scorecard internal tests to be based on the custom test format.  For this document, I’ve called this work, Phase 2, and is optional but worth consideration.  
+
+Today there are 2 internal scorecard tests that make use of the scorecard proxy, with these tests migrating to the custom test format, the scorecard proxy is likely not necessary which would also reduce the overall scorecard implementation complexity.
+
+#### Building the Proposed Design
+
+The proposed design was influenced greatly by the KUDO testing framework in various concepts, however, the proposed design doesn’t necessarily have to include KUDO as a dependency.  A decision of whether or not to incorporate KUDO testing APIs within the Red Hat solution needs to be reached.  Some pro’s and con’s to using KUDO testing frameworks are listed below.  If KUDO’s testing framework is not incorporated, then a custom built set of equivalent capabilities would need to be developed.
+The advantage of using [Kuttl](https://github.com/kudobuilder/kuttl) as the test framework in scorecard would include:
+
+ * Pre-built, works, community, evolving.
+ * Ready to work with, code exists and mostly works, a starting point exists.
+ * Handles the basic test case, that being to test Kube resource state
+ * Written in golang, making integration into scorecard a possibility.
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Upgrade / Downgrade Strategy
+
+To implement this custom test functionality, additions or changes to the existing scorecard configuration file and command line flags would be necessary and would not be supported in previous scorecard releases.
+
+## Implementation History
+
+ * Proposal initiated in March 2020.
+
+## Drawbacks
+
+Drawbacks of using Kuttl within this proposed design might include:
+
```

Comment:
I think we need to add here as drawbacks scenarios such as:  https://github.com/operator-framework/operator-sdk/issues/2579